kernel/vm.c

```diff
@@ -102,6 +102,27 @@ walk(pagetable_t pagetable, uint64 va, int alloc)
   return &pagetable[PX(0, va)];
 }
 
+void printwalk(pagetable_t pagetable, uint64 level)
+{
+  // there are 2^9 = 512 PTEs in a page table.
+  for(int i = 0; i < 512; i++){
```

Comment:
I recommend to add named constant for PTE's count to avoid magic numbers.

"A long descriptive name is better than a long descriptive comment."